sys/include/net/gnrc/ipv6/nib/nc.h

```diff
@@ -123,9 +123,181 @@ extern "C" {
+ *
+ * @param[in] entry     A neighbor cache entry.
+ *
+ * @return  The neighbor unreachability state of @p entry.
+ */
+static inline unsigned gnrc_ipv6_nib_nc_get_nud_state(const gnrc_ipv6_nib_nc_t *entry)
+{
+    return (entry->info & GNRC_IPV6_NIB_NC_INFO_NUD_STATE_MASK);
+}
+
+/**
+ * @brief   Gets state of neighbor unreachability state of an entry
+ *
+ * @param[in] entry     A neighbor cache entry.
+ *
+ * @return  true, if @p entry is a router.
+ * @return  false, if @p entry is not a router.
+ */
+static inline bool gnrc_ipv6_nib_nc_is_router(const gnrc_ipv6_nib_nc_t *entry)
+{
+    return (entry->info & GNRC_IPV6_NIB_NC_INFO_IS_ROUTER);
+}
+
+/**
+ * @brief   Gets interface from entry
+ *
+ * @param[in] entry     A neighbor cache entry
+ *
+ * @return  The interface identifier of @p entry.
+ * @return  0 if no interface is identified for @p entry.
+ */
+static inline unsigned gnrc_ipv6_nib_nc_get_iface(const gnrc_ipv6_nib_nc_t *entry)
+{
+    return (entry->info & GNRC_IPV6_NIB_NC_INFO_IFACE_MASK) >>
+           GNRC_IPV6_NIB_NC_INFO_IFACE_POS;
+}
+
+/**
+ * @brief   Gets address registration state of an entry
+ *
+ * @param[in] entry     A neighbor cache entry
+ *
+ * @return  The address registration state of @p entry.
+ */
+static inline unsigned gnrc_ipv6_nib_nc_get_ar_state(const gnrc_ipv6_nib_nc_t *entry)
+{
+    return (entry->info & GNRC_IPV6_NIB_NC_INFO_AR_STATE_MASK);
+}
+
+/**
+ * @brief   Adds an unmanaged neighbor entry to NIB
+ *
+ * @pre `(ipv6 != NULL) && (l2addr != NULL)`
+ * @pre `l2addr_len <= GNRC_IPV6_NIB_L2ADDR_MAX_LEN`
+ * @pre `(iface > KERNEL_PID_UNDEF) && (iface <= KERNEL_PID_LAST)`
+ *
+ * @param[in] ipv6          The neighbor's IPv6 address.
+ * @param[in] iface         The interface to the neighbor.
+ * @param[in] l2addr        The neighbor's L2 address.
+ * @param[in] l2addr_len    Length of @p l2addr.
+ *
+ * A neighbor cache entry created this way is marked as persistent.
+ * Also, a non-persistent neighbor or destination cache entry already in the
+ * NIB might be removed to make room for the new entry.
+ * If an entry pointing to the same IPv6 address as @p ipv6 exists already it
+ * will be overwritten and marked as unmanaged.
+ *
+ * If @ref GNRC_IPV6_NIB_CONF_ARSM != 0 @p l2addr and @p l2addr_len won't be set.
+ *
+ * @return  0 on success.
+ * @return  -ENOMEM, if no space is left in neighbor cache.
+ */
+int gnrc_ipv6_nib_nc_set(const ipv6_addr_t *ipv6, unsigned iface,
+                         const uint8_t *l2addr, size_t l2addr_len);
+
+/**
+ * @brief   Deletes neighbor with address @p ipv6 from NIB
+ *
+ * @pre `ipv6 != NULL`
+ *
+ * @param[in] ipv6 The neighbor's IPv6 address.
+ *
+ * If the @p ipv6 can't be found for a neighbor in the NIB nothing happens.
+ */
+void gnrc_ipv6_nib_nc_del(const ipv6_addr_t *ipv6);
+
+/**
+ * @brief   Mark neighbor with address @p ipv6 as reachable
+ *
+ * @pre `ipv6 != NULL`
+ *
+ * @param[in] ipv6 A neighbor's IPv6 address. May not be NULL.
```

Comment:
1. Huhh?
2. I prefer to mention this in the parameter doc as well explicitly for clarity.